### PR TITLE
Hardcode the default "snyk-secret"

### PR DESCRIPTION
**Pull request by sbose78**
* The name of the secret would always be `snyk-secret`
image

* The volume is set to 'optional: true' which means the absence of the `secret` wouldn't cause a failure ?

### CODE REVIEW

File: task/sast-snyk-check/0.1/sast-snyk-check.yaml [hunk 1]
```diff
@@ -16,7 +16,7 @@ spec:
   params:
     - name: SNYK_SECRET
       description: Name of secret which contains Snyk token.
-      default: ""
+      default: snyk-secret
```
Comment (enarha) [inline task/sast-snyk-check/0.1/sast-snyk-check.yaml:19]: maybe snyk-shared-secret?

**Comment by enarha** [comment]
I guess we can also remove the pipelines/core-services/snyk-secret.yaml patch?

**Comment by dirgim** [comment]
If we want to always attempt to run the `sast-synk-check` task, I think we will need to silence the error message that happens when trying to read the secret [here](https://github.com/redhat-appstudio/build-definitions/blob/c31f6ad4939f52b30b10331097a21a0d03ca1745/task/sast-snyk-check/0.1/sast-snyk-check.yaml#L45) because that seemed to be a point of confusion for users.

**Comment by sbose78** [comment]
@dirgim Maybe we should update the message here to reflect that this is an opt-in
"echo "SNYK_TOKEN not defined. Create $SNYK_SECRET secret with key containing Snyk token." | tee stdout.txt"

**Comment by sbose78** [comment]
This also means we'd have to update the name of the secret used by the services on infra-deployments ?

**Comment by sbose78** [comment]
To make things easier, I added the name of the secret "snyk-secret" in the message.

File: task/sast-snyk-check/0.1/sast-snyk-check.yaml [hunk 2]
```diff
@@ -44,7 +44,7 @@ spec:
         . /utils.sh
         SNYK_TOKEN="$(cat /etc/secrets/snyk_token)"
         if [[ -z $SNYK_TOKEN ]]; then
-          echo "SNYK_TOKEN not defined. Create $SNYK_SECRET secret with key containing Snyk token." | tee stdout.txt
+          echo "If you wish to use the Snyk code SAST task, please create a secret named snyk-secret with the key containing the Snyk token." | tee stdout.txt
```
Comment (Michkov) [inline task/sast-snyk-check/0.1/sast-snyk-check.yaml:47]: please provide exact key name.

**Review by Michkov — CHANGES_REQUESTED** [review]
Should be also https://github.com/redhat-appstudio/build-definitions/blob/a383e661bbf64c1500efc1cfe2b01f4936ec2a8b/pipelines/template-build/template-build.yaml#L53 updated or removed?

File: pipelines/template-build/template-build.yaml [hunk 3]
```diff
@@ -50,10 +50,6 @@ spec:
       name: java
       type: string
       default: "false"
-    - description: Snyk Token Secret Name
-      name: snyk-secret
-      type: string
-      default: ""
```
Comment (Michkov) [inline pipelines/template-build/template-build.yaml:56]: With deletion of this you need to delete references -> lines 203 and 213.

**Review by Michkov — APPROVED** [review]

**Review by dirgim — APPROVED** [review]
LGTM

**Pull request merged**